src/Pos/Launcher/Runner.hs

```diff
@@ -0,0 +1,465 @@
+import           Pos.Core.Timestamp          (timestampF)
+import           Pos.Crypto                  (createProxySecretKey, encToPublic)
+import           Pos.DB                      (MonadDB (..), runDBHolder)
+import           Pos.DB.DB                   (initNodeDBs, openNodeDBs)
+import           Pos.DB.GState               (getTip)
+import           Pos.DB.Misc                 (addProxySecretKey)
+import           Pos.Delegation.Holder       (runDelegationT)
+import           Pos.DHT.Model               (MonadDHT (..), converseToNeighbors,
+                                              getMeaningPart)
+import           Pos.DHT.Real                (KademliaDHTInstance,
+                                              KademliaDHTInstanceConfig (..),
+                                              runKademliaDHT, startDHTInstance,
+                                              stopDHTInstance)
+import           Pos.Launcher.Param          (BaseParams (..), LoggingParams (..),
+                                              NodeParams (..))
+import qualified Pos.Lrc.DB                  as LrcDB
+import           Pos.Slotting                (SlottingVar, mkNtpSlottingVar,
+                                              runNtpSlotting, runSlottingHolder)
+import           Pos.Ssc.Class               (SscConstraint, SscHelpersClass,
+                                              SscListenersClass, SscNodeContext,
+                                              SscParams, sscCreateNodeContext)
+import           Pos.Ssc.Extra               (ignoreSscHolder, mkStateAndRunSscHolder)
+import           Pos.Statistics              (getNoStatsT, runStatsT')
+import           Pos.Txp                     (runTxpHolder)
+import           Pos.Types                   (Timestamp (Timestamp))
+import qualified Pos.Update.DB               as GState
+import           Pos.Update.MemState         (runUSHolder)
+import           Pos.Util                    (mappendPair, runWithRandomIntervalsNow)
+import           Pos.Util.UserSecret         (usKeys)
+import           Pos.Worker                  (allWorkersCount)
+import           Pos.WorkMode                (MinWorkMode, ProductionMode, RawRealMode,
+                                              ServiceMode, StatsMode)
+
+data RealModeResources = RealModeResources
+    { rmTransport :: Transport
+    , rmDHT       :: KademliaDHTInstance
+    }
+
+----------------------------------------------------------------------------
+-- Service node runners
+----------------------------------------------------------------------------
+
+-- | Runs node as time-slave inside IO monad.
+runTimeSlaveReal
+    :: (SscListenersClass ssc, SscHelpersClass ssc)
+    => Proxy ssc -> RealModeResources -> BaseParams -> Production Timestamp
+runTimeSlaveReal sscProxy res bp = do
+    mvar <- liftIO newEmptyMVar
+    runServiceMode res bp (listeners mvar) outs . toAction $ \sendActions ->
+      case Const.isDevelopment of
```

Comment:
It seems that `Const.isDevelopment` is used to determine whether to do prod or dev behavior.